vpn/options.go

```diff
@@ -188,32 +192,43 @@ func parseRemoteOptions(tunnel *tunnelInfo, remoteOpts string) *tunnelInfo {
 				log.Println("bad mtu:", err)
 				continue
 			}
-			tunnel.mtu = mtu
+			t.mtu = mtu
 		}
 	}
-	return tunnel
+	return t
 }
 
-// I don't think I want to do much with the pushed options for now, other
-// than extracting the tunnel ip, but it can be useful to parse them into a map
-// and compare if there's a strong disagreement with the remote opts
-// XXX right now this only returns the ip. we could accept a tunnel struct and
-// write to it.
-func parsePushedOptions(pushedOptions []byte) string {
+// newTunnelInfoFromPushedOptions takes a map of string to array of strings, and returns
+// a new tunnel struct with the relevant info.
+func newTunnelInfoFromPushedOptions(opts map[string][]string) *tunnelInfo {
+	t := &tunnelInfo{}
+	gw := opts["route-gateway"]
+	if len(gw) >= 1 {
+		t.gw = gw[0]
+	}
+	ip := opts["ifconfig"]
+	if len(ip) >= 1 {
+		t.ip = ip[0]
+	}
+	return t
+}
+
+// pushedOptionsAsMap returns a map for the server-pushed options,
+// where the options are the keys and each space-separated value is the value.
```

Comment:
You should document that you never return a `nil` map

Reply:
done